plugins/uptimerobot/src/api.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ConfigApi, createApiRef, DiscoveryApi } from '@backstage/core';
+import { UptimerobotApi } from './types';
+
+export class FetchError extends Error {
```

Comment:
this should use the builtin standard for error (de)serialization https://github.com/backstage/backstage/blob/master/packages/errors/src/errors/ResponseError.ts#L60

and then rendered with this core component https://github.com/backstage/backstage/blob/master/packages/core/src/components/ResponseErrorPanel/ResponseErrorPanel.tsx#L161